codec-http2/src/test/java/io/netty/handler/codec/http2/Http2ConnectionRoundtripTest.java

```diff
@@ -702,6 +705,14 @@ protected void initChannel(Channel ch) throws Exception {
                         .validateHeaders(false)
                         .gracefulShutdownTimeoutMillis(0)
                         .build());
+                p.addLast(new ChannelInboundHandlerAdapter() {
+                    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+                        if (evt instanceof Http2ConnectionPrefaceWrittenEvent) {
+                            prefaceWrittenLatch.countDown();
+                            ctx.pipeline().remove(this);
```

Comment:
There is a call below which makes an assumption about what is the last item in the pipeline:

```java
http2Client = clientChannel.pipeline().get(Http2ConnectionHandler.class);
```

I could update this ... but it is not necessary if I just remove the handler. The handler is also not necessary after this user event is processed.